src/amo/components/HomepageShelves/index.js

```diff
@@ -0,0 +1,127 @@
+/* @flow */
+import * as React from 'react';
+import config from 'config';
+
+import LandingAddonsCard from 'amo/components/LandingAddonsCard';
+import LoadingText from 'amo/components/LoadingText';
+import {
+  INSTALL_SOURCE_FEATURED,
+  INSTALL_SOURCE_FEATURED_COLLECTION,
+  LANDING_PAGE_EXTENSION_COUNT,
+  LANDING_PAGE_THEME_COUNT,
+} from 'amo/constants';
+import translate from 'amo/i18n/translate';
+import type { ResultShelfType } from 'amo/reducers/home';
+import type { I18nType } from 'amo/types/i18n';
+
+import './styles.scss';
+
+type DefaultProps = {|
+  placeholderCount: number,
+|};
+
+type InternalProps = {|
+  i18n: I18nType,
+|};
+
+type Props = {|
+  ...DefaultProps,
+  ...InternalProps,
+  loading: boolean,
+  placeholderCount?: number,
+  shelves: Array<ResultShelfType>,
+|};
+
+class HomepageShelves extends React.Component<Props> {
+  /* eslint-disable react/no-array-index-key */
+  static defaultProps: DefaultProps = {
+    placeholderCount: LANDING_PAGE_EXTENSION_COUNT,
```

Comment:
I know I've reviewed this a few times now and this hasn't come up, but I just found myself wondering, why do we need this prop? This component will only ever be called from one place (the home page), so why does this need to be a prop? Could we just define this in here as a constant and be done with it?